dali/pipeline/graph_descr.cc

```diff
@@ -93,360 +95,428 @@ StorageDevice ParseStorageDevice(const std::string &io_device) {
+    new_node.parent_tensors.push_back(consumed_tensor_id);
 
     // Update the consumer info for this tensor
     TensorMeta meta;
-    meta.node = new_node->id;
+    meta.node = new_node.id;
     meta.index = i;
     meta.is_support = spec.IsArgumentInput(i);
     meta.storage_device = ParseStorageDevice(spec.InputDevice(i));
 
-    vector<TensorMeta> &consumer_info = tensor_consumers_[spec.Input(i)];
-    consumer_info.push_back(meta);
+    // Insert new tensor consumer
+    tensor_nodes_[consumed_tensor_id].consumers.push_back(meta);
   }
 
   // Mark this op as the source of its output tensors
+  // Create TensorNodes for outputs and respective edges for this OpNode -> TensorNodes
   for (int i = 0; i < spec.NumOutput(); ++i) {
-    string name = spec.Output(i);
-
     // Set the producer info for this tensor
     TensorMeta meta;
-    meta.node = new_node->id;
+    meta.node = new_node.id;
     meta.index = i;
     meta.is_support = spec.GetArgument<string>("device") == "support";
     meta.storage_device = ParseStorageDevice(spec.OutputDevice(i));
 
-    auto ret = tensor_producers_.insert({name, meta});
-    DALI_ENFORCE(ret.second, "Operator '" + spec.name() +
+    string name = spec.Output(i);
+
+    // Place new Tensor with producer info and add edge to OpNode.
+    auto &new_tensor = PlaceNewTensor();
+    new_tensor.producer = meta;
+    new_tensor.name = name;
+    new_node.children_tensors.push_back(new_tensor.id);
+
+    auto it_inserted = tensor_name_to_id_.insert({name, new_tensor.id});
+    DALI_ENFORCE(it_inserted.second, "Operator '" + spec.name() +
         "' has output with name " + name + ", but output "
         "with this name already exists as output of op '" +
-        this->node(TensorSourceID(name)).spec.name() + "'");
+        this->Node(TensorSourceID(name)).spec.name() + "'");
   }
 }
 
 void OpGraph::InstantiateOperators() {
   // traverse devices by topological order (support, cpu, mixed, gpu)
-  for (auto &node : support_nodes_) {
-    if (node.op)
-      continue;
-    auto &spec = node.spec;
-    string device = spec.GetArgument<string>("device");
-    node.op = SupportOperatorRegistry::Registry().Create(spec.name(), spec, &device);
-  }
-  for (auto &node : cpu_nodes_) {
-    if (node.op)
-      continue;
-    auto &spec = node.spec;
-    string device = spec.GetArgument<string>("device");
-    node.op = CPUOperatorRegistry::Registry().Create(spec.name(), spec, &device);
-  }
-  for (auto &node : mixed_nodes_) {
-    if (node.op)
-      continue;
-    auto &spec = node.spec;
-    string device = spec.GetArgument<string>("device");
-    node.op = MixedOperatorRegistry::Registry().Create(spec.name(), spec, &device);
-  }
-  for (auto &node : gpu_nodes_) {
-    if (node.op)
-      continue;
-    auto &spec = node.spec;
-    string device = spec.GetArgument<string>("device");
-    node.op = GPUOperatorRegistry::Registry().Create(spec.name(), spec, &device);
+  OpType order[] = { OpType::SUPPORT, OpType::CPU,
```

Comment:
Would the line exceed 100 chars if it were not broken?

Reply:
Done.